src/wheredataframe/wheredataframe.jl

```diff
@@ -0,0 +1,287 @@
+"""
+    WhereDataFrame{<:AbstractDataFrame,<:AbstractIndex}
+
+    The result of a [`where`](@ref) operation on an `AbstractDataFrame`; a
+    subset of a `AbstractDataFrame`
+    Not meant to be constructed directly, see `where`.
+"""
+struct WhereDataFrame{D<:AbstractDataFrame, T<:AbstractVector{Int}}
+    parent::D
+    rows::T
+end
+
+function WhereDataFrame(parent::AbstractDataFrame, rows::AbstractVector{Bool})
+    if length(rows) != nrow(parent)
+        throw(ArgumentError("invalid length of `AbstractVector{Bool}` row index" *
+                            " (got $(length(rows)), expected $(nrow(parent)))"))
+    end
+    return WhereDataFrame(parent, findall(rows))
+end
+
+rows(wdf::WhereDataFrame) = getfield(wdf, :rows)
+Base.parent(wdf::WhereDataFrame) = getfield(wdf, :parent)
+
+
+"""
+    where(d::AbstractDataFrame, args...)
+
+`where` introduces a `where` clause, which will be applied in the next function. Returns a `WhereDataFrame`.
+
+`args...` obey the same syntax as `select(d, args...)`
+Rows that return missing are understood as false
+
+- `filter`/`filter!` returns an AbstractDataFrame after filtering (resp. deleting) specified rows
```

Comment:
please add that this is a list of functions that support `where` wrapper. In general - in the long term almost all functions we define could support it I think (but we can make this change gradually)

Reply:
in particular you omit `select`/`select!` now and e.g. `insertcols!`.